pyccel/codegen/printing/ccode.py

```diff
@@ -1120,42 +1118,115 @@ def _print_Assign(self, expr):
-                arg = ', '.join(self._print(i) for i in arg)
-                dummy_array = "%s %s[] = {%s};\n" % (declare_dtype, dummy_array_name, arg)
-                cpy_data = "memcpy({0}.{2}, {1}, {0}.buffer_size);".format(lhs, dummy_array_name, dtype)
-                return  '%s%s\n' % (dummy_array, cpy_data)
-
-        if isinstance(rhs, (NumpyFull)):
-            code_init = ''
-            if rhs.fill_value is not None:
-                if isinstance(rhs.fill_value, Literal):
-                    dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
-                    code_init = 'array_fill(({0}){1}, {2});'.format(dtype, self._print(rhs.fill_value), lhs)
-                else:
-                    code_init = 'array_fill({0}, {1});'.format(self._print(rhs.fill_value), lhs)
+            return '%s\n' % (cpy_data)
+        else :
+            arg = ', '.join(self._print(i) for i in arg)
+            dummy_array = "%s %s[] = {%s};\n" % (declare_dtype, dummy_array_name, arg)
+            if expr.lhs.is_stack_array:
+                cpy_data = self._init_stack_array(expr, dummy_array_name)
             else:
-                return ''
-            return '{}\n'.format(code_init)
+                cpy_data = "memcpy({0}.{2}, {1}, {0}.buffer_size);".format(self._print(lhs), dummy_array_name, dtype)
+            return  '%s%s\n' % (dummy_array, cpy_data)
 
-        rhs = self._print(rhs)
-        return '{} = {};'.format(lhs, rhs)
+    def arrayFill(self, expr):
+        """ print the Assignement of a NdArray
+
+        parameters
+        ----------
+            expr : PyccelAstNode
+                The Assign Node used to get the lhs and rhs
+        Return
+        ------
+            String
+                Return a str that contain a call to the C function array_fill,
+                if the ndarray is a stack_array the str will contain the initialization
+        """
+        rhs = expr.rhs
+        lhs = expr.lhs
+        code_init = ''
+        if lhs.is_stack_array:
+            declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
+            length = '*'.join(self._print(i) for i in lhs.shape)
+            buffer_array = "({declare_dtype}[{length}]){{}}".format(declare_dtype = declare_dtype, length=length)
+            code_init += self._init_stack_array(expr, buffer_array)
+        if rhs.fill_value is not None:
+            if isinstance(rhs.fill_value, Literal):
+                dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
+                code_init += 'array_fill(({0}){1}, {2});\n'.format(dtype, self._print(rhs.fill_value), self._print(lhs))
+            else:
+                code_init += 'array_fill({0}, {1});\n'.format(self._print(rhs.fill_value), self._print(lhs))
+        return '{}'.format(code_init)
+
+    def _init_stack_array(self, expr, buffer_array):
+        """ return a string that have handle the assign of a stack ndarray
```

Comment:
Grammar `return a string that have handle the assign of a stack ndarray`->`return a string which handles the assignment of a stack ndarray`